sql/core/src/main/scala/org/apache/spark/sql/jdbc/PostgresDialect.scala

```diff
@@ -404,4 +404,26 @@ private case class PostgresDialect()
       case _ =>
     }
   }
+
+  override def compileExpression(expr: Expression): Option[String] = {
+    val builder = new PostgresSQLBuilder()
+    try {
```

Comment:
perhaps its better to only override visitBinaryArithmetics?

We have similar problem with some of the functions and we use `dialectFunctionName` to translate from Spark to local dialect

Reply:
+1 here, let's override last possible method in chain of execution

Reply:
done